src/adm/tiles.rs

```diff
@@ -172,39 +178,64 @@ pub async fn get_tiles(
-            .get("fake-response")
-            .unwrap_or(&default)
-            .to_str()
-            .unwrap()
-            .to_owned();
-        trace!("Getting fake response: {:?}", &test_response);
-        AdmTileResponse::fake_response(&state.settings, test_response)?
-    } else {
-        state
-            .reqwest_client
-            .get(adm_url)
-            .timeout(Duration::from_secs(settings.adm_timeout))
-            .send()
-            .await
-            .map_err(|e| {
-                // ADM servers are down, or improperly configured
-                let mut err: HandlerError = HandlerErrorKind::AdmServerError().into();
-                err.tags.add_extra("error", &e.to_string());
-                err
-            })?
-            .error_for_status()?
-            .json()
-            .await
-            .map_err(|e| {
-                // ADM servers are not returning correct information
-                HandlerErrorKind::BadAdmResponse(format!("ADM provided invalid response: {:?}", e))
-            })?
+    let response: AdmTileResponse = match state.settings.test_mode {
+        crate::settings::TestModes::TestFakeResponse => {
+            let default = HeaderValue::from_str(DEFAULT).unwrap();
+            let test_response = headers
+                .unwrap_or(&HeaderMap::new())
+                .get("fake-response")
+                .unwrap_or(&default)
+                .to_str()
+                .unwrap()
+                .to_owned();
+            trace!("Getting fake response: {:?}", &test_response);
+            AdmTileResponse::fake_response(&state.settings, test_response)?
+        }
+        crate::settings::TestModes::TestTimeout => {
+            trace!("### Timeout!");
+            return Err(HandlerErrorKind::AdmLoadError().into());
+        }
+        _ => {
+            state
+                .reqwest_client
+                .get(adm_url)
+                .timeout(Duration::from_secs(settings.adm_timeout))
+                .send()
+                .await
+                .map_err(|e| {
+                    // If we're just starting up, we're probably swamping the partner servers as
+                    // we fill the queue. Instead of returning a normal 500 error, let's
+                    // return something softer to keep our SRE's blood pressure lower.
+                    //
+                    // We still want to track this as a server error later.
+                    //
+                    // TODO: Remove this after the shared cache is implemented.
+                    let mut err: HandlerError = if e.is_timeout()
+                        && Instant::now()
+                            .checked_duration_since(state.start_up)
+                            .unwrap_or_else(|| Duration::from_secs(0))
+                            <= Duration::from_secs(state.settings.adm_timeout)
+                    {
+                        HandlerErrorKind::AdmLoadError().into()
```

Comment:
I'm not totally convinced we need all this because we don't seem to generate too many `AdmServerError`s AFAICT but I'm ok keeping it and seeing its actual usage. 

Though I'm thinking `AdmLoadError` should emit a metric instead of going to sentry (just needs a line added to `HandlerErrorKind::metric_label`) since we know exactly what causes it. I'd say the the same for `AdmServerError` (at least when `e.is_timeout()` which is the majority of its cases) but let's worry about that later.

Reply:
The reason I think `AdmLoadError` should be a sentry event is because it's both significant and actionable, it's just that during the initial startup, there can be a lot of false events because we're swamping the sources. If we make it a metric, we're less inclined to take action. I am not going to make this a strong stand, though, so if we see different behaviors in the future, I'm find converting this to a metric.